addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/internal/handler/RouteHandler.java

```diff
@@ -0,0 +1,292 @@
+                    }
+                    break;
+                default:
+                    logger.warn("Unhandled command for channel: {} with comamnd: {}", channelUID.getId(), command);
+            }
+        } else {
+            logger.warn("The OneBusAway binding is a read-only binding and can not handle commands.");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleUpdate(ChannelUID channelUID, State newState) {
+        logger.warn("The OneBusAway binding is a read-only binding and can not handle channel updates.");
+        super.handleUpdate(channelUID, newState);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        logger.debug("Initializing OneBusAway route stop...");
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Checking configuration...");
+
+        config = loadAndCheckConfiguration();
+        if (config == null) {
+            logger.debug("Initialization of OneBusAway route stop failed!");
+            return;
+        }
+        StopHandler stopHandler = getStopHandler();
+        if (stopHandler != null) {
+            // We will be marked as ONLINE when we get data in our callback, which should be immediately because the
+            // StopHandler, our bridge, won't be marked as online until it has data itself.
+            stopHandler.registerRouteDataListener(this);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Bridge unavailable");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dispose() {
+        cancelAllScheduledFutures();
+        StopHandler stopHandler = getStopHandler();
+        if (stopHandler != null) {
+            stopHandler.unregisterRouteDataListener(this);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getRouteId() {
+        return config.getRouteId();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void onNewRouteData(long lastUpdateTime, List<ArrivalAndDeparture> data) {
+        if (data.isEmpty()) {
+            return;
+        }
+        // Publish to all of our linked channels.
+        Calendar now = Calendar.getInstance();
+        for (Channel channel : getThing().getChannels()) {
+            if (channel.getKind() == ChannelKind.TRIGGER) {
+                scheduleTriggerEvents(channel.getUID(), now, data);
+            } else {
+                publishChannel(channel.getUID(), now, lastUpdateTime, data);
+            }
+        }
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    private StopHandler getStopHandler() {
+        Bridge bridge = getBridge();
+        if (bridge == null || !(bridge.getHandler() instanceof StopHandler)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "No bridge available");
```

Comment:
Note:
1. getBridge() should always return you an instance, so it is unlikely that we get here
2. dispose() must NOT call updateStatus, but as it calls getStopHandler it potentially might do this. So you should remove this line here.